query-planner-js/src/buildPlan.ts

```diff
@@ -256,6 +257,21 @@ class QueryPlanningTaversal<RV extends Vertex> {
     return FetchDependencyGraph.create(this.subgraphs);
   }
 
+  // Moves the first closed branch to after any branch having more options.
+  private reorderFirstBranch() {
+    const firstBranch = this.closedBranches[0];
+    let i = 1;
+    while (i < this.closedBranches.length && this.closedBranches[i].length > firstBranch.length) {
+      i++;
+    }
+    // `i` is the smallest index of an element having the same number or less options than the first one,
+    // so we switch that first branch with the element "before" `i` (which has more elements). 
+    if (i > 0) {
```

Comment:
This if statement doesn't do anything since i will always be greater than 1.

Reply:
Indeed, force of habit of starting iterations at 0. Removed it.

Reply:
My bad, didn't notice you were swapping with `i-1` instead of `i`